### DISCUSSION TITLE
Pyright configuration not honored by zed?

### DISCUSSION THREAD

**Discussion by somedude72** [zed-industries/zed#14424] [Feature Requests]
I have the following in my `settings.json`:
```json
"lsp": {
    "pyright": {
        "settings": {
            "python.analysis.typeCheckingMode": "basic",
            "python.analysis.diagnosticSeverityOverrides": {
                "reportReturnType": "warning",
                "reportArgumentType": "warning",
                "reportOperatorIssue": "warning",
                "reportImportCycles": "warning",
                "reportDuplicateImport": "warning"
            },

            "python.venvPath": ".venv/bin/python"
        }
    }
}
```

However, zed/pyright does not seem to have taken note of any of this, because the language server can't seem to find packages installed in the project venv folder, and still shows all of the overridden diagnostics as the "error" severity. How can I fix this?

**Comment by Cernunnos** [comment]
Create a `pyrightconfig.json` (in the root folder of the project) with the following content:
```json
{
  "typeCheckingMode": "basic",
  "reportReturnType": "warning",
  "reportArgumentType": "warning",
  "reportOperatorIssue": "warning",
  "reportImportCycles": "warning",
  "reportDuplicateImport": "warning",
  "venvPath": ".",
  "venv": ".venv"
}
```

**Reply by somedude72** [reply]
That only works if I setup each python project or file that I plan to open with a `pyrightconfig.json`. Is there a method that allows me to set this up globally without having to worry about individual projects? 

By the way, the zed docs for configuring the python lsp actually has quite a [similar code snippet](https://zed.dev/docs/languages/python#settings). Unfortunately, copy and pasting that into my `settings.json` seems to have no effect on anything either.

**Comment by drvdana** [comment]
Based on this documentation: https://zed.dev/docs/configuring-languages#nested-objects, you can't have dot-delimited strings. However, I changed my pyright settings to be all nested objects, and I'm still not seeing pyright configuration changes reflected when analyzing my project.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
